graph/tarjan.cpp

```diff
@@ -0,0 +1,227 @@
+#include<iostream> /// for general input and output
+#include<vector> /// for storing graph
+#include<queue> /// used for BFS algorithm in below namespace
+#include<list>  /// used for making adjacency list representation of graph
+#include<set>   /// for returning result of tarjan's agorithm
+
+/**
+ * @namespace graph
+ * @brief Graph algorithms
+ */
+namespace graph {
+class Graph
+{
+public:
+
+    std::vector<std::list<std::pair<int, int>>> g;
+    int vertex;
+    bool isDirected=false;
+    int flag=0;
+    std::vector<std::pair<int,int>> allEdges; // for keeping track of all edges used for cycle detection.
+    std::vector<std::pair<int,std::pair<int,int> >> allWEdges; //it stores weight and vertex of edges.
+
+
+    explicit Graph(int default_vertex = 10) 
```

Comment:
Functions should be documented explaining the use of its parameters, a brief description, and what it returns.